hpx/runtime/actions/plain_action.hpp

```diff
@@ -169,6 +169,41 @@ namespace hpx { namespace traits
 
 /// \endcond
 
+///////////////////////////////////////////////////////////////////////////////
+/// \def HPX_DECLARE_PLAIN_ACTION(func, name)
+/// \brief Declares a plain action type
+///
+#define HPX_DECLARE_PLAIN_ACTION(...)                                         \
+    HPX_DECLARE_PLAIN_ACTION_(__VA_ARGS__)                                    \
+    /**/
+
+/// \cond NOINTERNAL
+
+#define HPX_DECLARE_PLAIN_DIRECT_ACTION(...)                                  \
+    HPX_DECLARE_PLAIN_DIRECT_ACTION_(__VA_ARGS__)                             \
+    /**/
+
+#define HPX_DECLARE_PLAIN_ACTION_(...)                                        \
+    HPX_UTIL_EXPAND_(BOOST_PP_CAT(                                            \
+        HPX_DECLARE_PLAIN_ACTION_, HPX_UTIL_PP_NARG(__VA_ARGS__)              \
+    )(__VA_ARGS__))                                                           \
+    /**/
+
+#define HPX_DECLARE_PLAIN_DIRECT_ACTION_(...)                                 \
+    HPX_UTIL_EXPAND_(BOOST_PP_CAT(                                            \
+        HPX_DECLARE_PLAIN_DIRECT_ACTION_, HPX_UTIL_PP_NARG(__VA_ARGS__)       \
+    )(__VA_ARGS__))                                                           \
+    /**/
+
```

Comment:
Am 29.04.2016 9:32 nachm. schrieb "Hartmut Kaiser" <notifications@github.com
>:
>
> In hpx/runtime/actions/plain_action.hpp:
>
> > +#define
HPX_DECLARE_PLAIN_DIRECT_ACTION(...)                                  \
> > +
HPX_DECLARE_PLAIN_DIRECT_ACTION_(__VA_ARGS__)                             \
> > +    /**/
> > +
> > +#define
HPX_DECLARE_PLAIN_ACTION_(...)                                        \
> > +
HPX_UTIL_EXPAND_(BOOST_PP_CAT(                                            \
> > +        HPX_DECLARE_PLAIN_ACTION_,
HPX_UTIL_PP_NARG(__VA_ARGS__)              \
> > +
)(__VA_ARGS__))                                                           \
> > +    /**/
> > +
> > +#define
HPX_DECLARE_PLAIN_DIRECT_ACTION_(...)                                 \
> > +
HPX_UTIL_EXPAND_(BOOST_PP_CAT(                                            \
> > +        HPX_DECLARE_PLAIN_DIRECT_ACTION_,
HPX_UTIL_PP_NARG(__VA_ARGS__)       \
> > +
)(__VA_ARGS__))                                                           \
> > +    /**/
> > +
>
> Doesn't it expand to HPX_DECLARE_PLAIN_ACTION_N, where N ==
HPX_UTIL_PP_NARG(__VA_ARGS__)?

I read _DIRECT_ in there.

>
> —
> You are receiving this because you commented.
>
> Reply to this email directly or view it on GitHub